vk::UniqueInstance can not be default-constructed

This program fails to compile due to an ambiguous constructor in ObjectDestroy.

```
#include "vulkan/vulkan.hpp"

int main(int argc, char** argv) {

    vk::UniqueInstance instance;

    return 0;
}
```

The error (clang++ 6.0.0):
```
In file included from test.cpp:2:
/usr/include/vulkan/vulkan.hpp:598:9: error: call to constructor of 'UniqueHandleTraits<Instance, DispatchLoaderStatic>::deleter' (aka
      'ObjectDestroy<vk::NoParent, vk::DispatchLoaderStatic>') is ambiguous
      : Deleter()
        ^
test.cpp:6:24: note: in instantiation of member function 'vk::UniqueHandle<vk::Instance, vk::DispatchLoaderStatic>::UniqueHandle' requested here
    vk::UniqueInstance instance;
                       ^
/usr/include/vulkan/vulkan.hpp:2689:7: note: candidate constructor
      ObjectDestroy()
      ^
/usr/include/vulkan/vulkan.hpp:2694:7: note: candidate constructor
      ObjectDestroy( Optional<const AllocationCallbacks> allocationCallbacks = nullptr, Dispatch const &dispatch = VULKAN_HPP_DEFAULT_DISPATCHER ) VULKAN_HPP_NOEXCEPT
      ^
1 error generated.
```

It was fixed by removing the default allocation callback argument for the vk::NoParent specialization of ObjectDestroy:
```
// Does not work
ObjectDestroy( Optional<const AllocationCallbacks> allocationCallbacks = nullptr, Dispatch const &dispatch = VULKAN_HPP_DEFAULT_DISPATCHER ) VULKAN_HPP_NOEXCEPT
        : m_allocationCallbacks( allocationCallbacks )
        , m_dispatch( &dispatch )
      {}
// Works
ObjectDestroy( Optional<const AllocationCallbacks> allocationCallbacks, Dispatch const &dispatch = VULKAN_HPP_DEFAULT_DISPATCHER ) VULKAN_HPP_NOEXCEPT
        : m_allocationCallbacks( allocationCallbacks )
        , m_dispatch( &dispatch )
      {}
```

--- a/VulkanHppGenerator.cpp
+++ b/VulkanHppGenerator.cpp
@@ -4629,7 +4629,7 @@ int main( int argc, char **argv )
         , m_dispatch( nullptr )
       {}
 
-      ObjectDestroy( Optional<const AllocationCallbacks> allocationCallbacks = nullptr, Dispatch const &dispatch = VULKAN_HPP_DEFAULT_DISPATCHER ) VULKAN_HPP_NOEXCEPT
+      ObjectDestroy( Optional<const AllocationCallbacks> allocationCallbacks, Dispatch const &dispatch = VULKAN_HPP_DEFAULT_DISPATCHER ) VULKAN_HPP_NOEXCEPT
         : m_allocationCallbacks( allocationCallbacks )
         , m_dispatch( &dispatch )
       {}

--- a/vulkan/vulkan.hpp
+++ b/vulkan/vulkan.hpp
@@ -2691,7 +2691,7 @@ namespace VULKAN_HPP_NAMESPACE
         , m_dispatch( nullptr )
       {}
 
-      ObjectDestroy( Optional<const AllocationCallbacks> allocationCallbacks = nullptr, Dispatch const &dispatch = VULKAN_HPP_DEFAULT_DISPATCHER ) VULKAN_HPP_NOEXCEPT
+      ObjectDestroy( Optional<const AllocationCallbacks> allocationCallbacks, Dispatch const &dispatch = VULKAN_HPP_DEFAULT_DISPATCHER ) VULKAN_HPP_NOEXCEPT
         : m_allocationCallbacks( allocationCallbacks )
         , m_dispatch( &dispatch )
       {}

